Sources/UIAlertControllerView.swift

```diff
@@ -43,48 +57,103 @@ class UIAlertControllerView: UIView {
             return button
         }
 
+        clipsToBounds = true
         backgroundColor = .white
         layer.cornerRadius = 3
 
-        header.sizeToFit()
-        header.frame.height += verticalPadding
-        header.layer.contentsGravity = .top
         addSubview(header)
+        text.map { addSubview($0) }
+        buttons.forEach { addSubview($0) }
+
+        subviewWidth = {
+            let minWidth: CGFloat = 300
+            let maxWidth: CGFloat
+            if UIScreen.main.isPortrait {
+                maxWidth = UIScreen.main.bounds.width - 4 * horizontalPadding
+            } else {
+                maxWidth = max(minWidth, UIScreen.main.bounds.width / 1.5)
+            }
 
-        buttons.forEach{ addSubview($0) }
+            subviews.forEach { $0.sizeToFit() }
+            guard let largestSubviewWidth = subviews.map({ $0.frame.width }).max() else {
+                assertionFailure("no subviews exist to determine largestSubviewWidth")
+                return minWidth
+            }
+            return min(maxWidth, max(largestSubviewWidth, minWidth))
+        }()
     }
 
     override func layoutSubviews() {
-        switch style {
-        case .actionSheet: layoutAsActionSheet()
-        default: assertionFailure(
-            "The UIAlertControllerStyle style, \(style), is not implemented yet!")
+        subviews.forEach { $0.sizeToFit() }
```

Comment:
Not 100% sure - but I think this might not be needed.

Reply:
hmmm yeah this is kind of tricky. it shouldn't be necessary but it currently is for the layout to look correct. will investigate this.

Reply:
(done, wasn't that complicated in the end)